### PR TITLE
GG-34220 Support for JDK17

### PR DESCRIPTION
**Pull request by SammyVimes**

### CODE REVIEW

**Review by ibessonov — APPROVED** [review]

File: modules/tools/pom.xml [hunk 1]
```diff
@@ -37,13 +37,23 @@
         <profile>
             <id>java-9+</id>
```
Comment (rpuch) [inline modules/tools/pom.xml:38]: Do we still want to call this profile `java-9+` (as it is not going to include 15+ anymore)? Wouldn't something like `java-9-14` more honest? Or do we want to be 'backward-compatible' with profile naming?

File: parent/pom.xml [hunk 2]
```diff
@@ -1203,7 +1203,7 @@
         <profile>
             <id>java-9+</id>
```
Comment (rpuch) [inline parent/pom.xml:1204]: Same question about profile naming as above

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/filehandle/FileWriteHandleImpl.java [hunk 3]
```diff
@@ -611,4 +586,137 @@ public String safePosition() {
...
+                force.invoke(mappedMemoryUtils, fd.get(buf), address, isSync, index, len);
+            }
+            catch (IllegalAccessException | InvocationTargetException e) {
+                throw new IgniteCheckedException(e);
+            }
+        }
+    }
+
+    /**
+     * Runs fsync on pre-java15 JVMs that doesn't offer a possibility to fsync mapped byte buffers in an aligned way.
```
Comment (rpuch) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/filehandle/FileWriteHandleImpl.java:682]: Probably, 'that don't offer'?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/filehandle/FileWriteHandleImpl.java [hunk 4]
```diff
@@ -611,4 +586,137 @@ public String safePosition() {
...
+        if (javaVersion >= 15)
+            return new JDK15FSyncer();
+
+        return new LegacyFSyncer();
+    }
+
+    /**
+     * Runs fsync operation on JRE15 and higher using MappedMemoryUtils which provides aligned fsync.
+     */
+    private static class JDK15FSyncer implements MMapFSyncer {
```
Comment (rpuch) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/filehandle/FileWriteHandleImpl.java:623]: Does this syncer perform better than the legacy one? Its code does not seem much simpler, but now we have 2 syncers to support.

Comment (SammyVimes) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/filehandle/FileWriteHandleImpl.java:623]: The old one doesn't work on JDK15 and newer

Comment (rpuch) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/filehandle/FileWriteHandleImpl.java:623]: Oh, the most obvious one :)

**Review by rpuch — APPROVED** [review]

**Pull request merged**